src/a-runtime-console/kubernetes/store/oauth-config-store.spec.ts

```diff
@@ -0,0 +1,283 @@
+
+import { NotificationsService } from 'app/shared/notifications.service';
+
+describe('OauthConfigStore', () => {
+
+  let mockUserService: UserService;
+  let mockBackend: MockBackend;
+  let oauthStore: OAuthConfigStore;
+
+  let mockLogger: jasmine.SpyObj<Logger>;
+  let mockErrorHandler: jasmine.SpyObj<ErrorHandler>;
+  let mockNotificationsService: jasmine.SpyObj<NotificationsService>;
+
+  let user: User = {
+    attributes: {
+      fullName: 'mock',
+      imageURL: 'mock',
+      username: 'mock',
+      cluster: 'http://api.example.com/cluster/'
+    },
+    id: 'mock',
+    type: 'mock'
+  };
+
+  let data = {};
+
+  let subscription: Subscription;
+
+  beforeEach(() => {
+    mockLogger = jasmine.createSpyObj<Logger>('Logger', ['error']);
+    mockErrorHandler = jasmine.createSpyObj<ErrorHandler>('ErrorHandler', ['handleError']);
+    mockNotificationsService = jasmine.createSpyObj<NotificationsService>('NotificationsService', ['message']);
+  });
+
+  describe('success state', () => {
+    beforeEach(() => {
+      mockUserService = createMock(UserService);
+      mockUserService.loggedInUser = Observable.of(user).publish();
+
+      TestBed.configureTestingModule({
+        imports: [HttpModule],
+        providers: [
+          {
+            provide: XHRBackend, useClass: MockBackend
+          },
+          {
+            provide: UserService, useClass: mockUserService
+          },
+          {
+            provide: Logger, useValue: mockLogger
+          },
+          {
+            provide: ErrorHandler, useValue: mockErrorHandler
+          },
+          {
+            provide: NotificationsService, useValue: mockNotificationsService
+          },
+          {
+            // provide OAuthConfigStore with a factory inside the fakeAsync zone
+            // so the mockBackend can catch http requests made inside the constructor
+            provide: OAuthConfigStore, useFactory: fakeAsync((
+              http: Http,
+              mockBackend: MockBackend,
+              logger: Logger,
+              errorHandler: ErrorHandler,
+              notifications: NotificationsService
+            ) => {
+              mockBackend.connections.subscribe((connection: MockConnection) => {
+                connection.mockRespond(new Response(new ResponseOptions({
+                  body: JSON.stringify(data),
+                  status: 200
+                })));
+              });
+              return new OAuthConfigStore(http, mockUserService, logger, errorHandler, notifications);
+            }),
+            deps: [Http, XHRBackend, Logger, ErrorHandler, NotificationsService]
+          }
+        ]
+      });
+
+      mockBackend = TestBed.get(XHRBackend);
+      oauthStore = TestBed.get(OAuthConfigStore);
+    });
+
+    it('should load and set latest oauthconfig on init', (done: DoneFn) => {
+      mockUserService.loggedInUser.connect();
+
+      oauthStore.loading.subscribe((val: boolean) => {
+        if (!val) {
```

Comment:
Does this `if (!val)` check need to be present in this test? The truthiness of `val` should be well known in this test case, and the test is geared to fail by timeout if it's wrong. Perhaps the check can be removed and a `expect().toBeFalsey()` can be added instead, so that if it is *not* as expected the reason for the test failure is more explicitly reported than a vague timeout?

Reply:
I think I did these if statements for resource && loading because they're both BehaviorSubjects so I expected them to emit their initial value to the subscriber and eventually emit the value I want. Looks like I was wrong though; fixed.